src/Parallel/CharmMain.cpp

```diff
@@ -19,6 +19,23 @@ namespace charmxx {
 template <class...>
 using void_t = void;
 
+template <class T>
```

Comment:
this looks general; should it be in Utilities instead of Parallel/CharmMain?

Reply:
It could be in theory, I'm not sure if it will be needed elsewhere in practice, though. This is fairly specific to having to register functions with Charm++, in other use cases type deduction will work fine.